Feature/josh action23 print table

JoshRupp opened the pull request:
Please review my  printTables.py file. This will help print a formatted table and organize output in an appealing way.  If any concerns please leave a comment.  Advancements are inevitable.

EnderStir commented on printTables.py:
```diff
@@ -0,0 +1,247 @@
+        self.intersectionChar = "+"
+        self.outputString = ""
+        self.tableSize = len(self.table)
+        self.headerSize = 0
```
headerSize isn't very clear as to what this is tracking. It should be something like: numOfColumns

JoshRupp replied:
That does seem like a better name. Potentially I will change that. However, I feel that this entire class should be deprecated.  I have implemented a better version of a table.

EnderStir commented on printTables.py:
```diff
@@ -0,0 +1,247 @@
+
+
+
+    def getHeaderSize(self):
```
This function isn't following the convention of 'get' functions. 'get' functions should return the value, where this is calculating the value. So, maybe this should be a 'find', or a 'calculate' function, something like 'findHeaderSize', or 'findNumOfColumns'.

JoshRupp replied:
Good point.

EnderStir commented on printTables.py:
```diff
@@ -0,0 +1,247 @@
+    """
+    def finalSymError (self):
+
+        return self.maxWordLen - 18
```
Why is this -18?

JoshRupp replied:
The math seemed to work as long as the table was a perfect rectangle. Unfortunately, in life, and in programming inputs are not ever perfect.  your right. This is a great question.

EnderStir commented on printTables.py:
```diff
@@ -0,0 +1,247 @@
+    """
+    def resizing (self):
+
+        return self.maxWordLen * 5 - 3
```
Why is this * 5 - 3? Is 5 the number of columns?

JoshRupp replied:
This math also was an illusion. Seemed to work initially, but with data that is not perfectly  symmetrical it fails.

EnderStir commented on printTables.py:
```diff
@@ -0,0 +1,247 @@
+    array and add n to it. Then set this value to the class member self.maxWordLen.  In so doing, you can organically 
+    create a column width to fit the needs of any table that is of a matrix 2D array type of data structure. 
+    """
+    def organicCOLCalculation(self, n):
```
This function name should be more clear. Why is it 'organic'?

JoshRupp replied:
Because it is supposed to loop though and find the longest word in a table. For any table type passed to it. Therefore, organic Column calculation means find the next longest work in a table passed and recalculate it on the  fly.  To bad the rest of the mathematics was elementary at best. Otherwise this theoretically could have worked out.

EnderStir commented on printTables.py:
```diff
@@ -0,0 +1,247 @@
+    Function truncates a world if necessary or appends space characters if necessary to create a 
+    dynamic right justified look when the table is printed and its contents are presented. 
+    """
+    def computeWordLength(self, text):
```
This function name sounds like it should be returning the lenght of a word, when this function is actually truncating or extending. Can you make this function name more clear?

JoshRupp replied:
Sure

EnderStir commented on printTables.py:
```diff
@@ -0,0 +1,247 @@
+     Function is designed to multiply the member variable self.boarder to a specified number 
+     This function helps create the character designed boarder
+    """
+    def mulWordWithrowChar(self, num):
```
I think this function can be updated to use maxRowWidth

JoshRupp replied:
Ok. Yea this whole implementation was not really well thought out.  I have created a second version which seems more flexible and stable.

EnderStir commented on commands.py:
```diff
@@ -0,0 +1,119 @@
+
```
This file should be removed.

EnderStir reviewed (changes requested):
When I add or remove a column from the table it doesn't display correctly. The top/bottom lines are either too long or two short.

JoshRupp committed:
updated this document via a manual merge conflict. should be good now

JoshRupp committed:
Merge branch 'jarDesignCEG4110' of https://github.com/EnderStir/ceg4110-group-12-spring-2021 into feature/Josh-Action23-printTable

# Conflicts:
#	Information/Design.md

JoshRupp committed:
Created a new implementation to printing a formatted table. With the
semester drawing to a close and my CEG 4350 project still not done. I
to cheat a little and use a python library to aid in my implementation
for printing a table.

JoshRupp committed:
updated version of pTables.py with added features. improved features are
sorting a table by any header specified. Moreover, you can obtain the
string variable representation of the table, or print the object version
of the table.

JoshRupp committed:
updated class name as instructed. Altered variable names to more of the
python way.

JoshRupp committed:
Merge branch 'Feature/josh-action23-print-tables-revision' of https://github.com/EnderStir/ceg4110-group-12-spring-2021 into feature/Josh-Action23-printTable

JoshRupp committed:
Merge branch 'develop' of https://github.com/EnderStir/ceg4110-group-12-spring-2021 into feature/Josh-Action23-printTable

# Conflicts:
#	Information/Design.md

JoshRupp committed:
implemented new way of printing tables with a new class.
